src/cider/nrepl/middleware/stacktrace.clj

```diff
@@ -156,13 +157,16 @@
 ;;; ## Middleware
 
 (defn wrap-stacktrace-reply
-  [{:keys [session transport pprint-fn] :as msg}]
+  [{:keys [session transport pprint-fn storage-key] :as msg}]
   ;; no stacktrace support for cljs currently - they are printed by piggieback anyway
-  (if-let [e (and (not (cljs/grab-cljs-env msg))
-                  (@session #'*e))]
-    (doseq [cause (analyze-causes e pprint-fn)]
-      (t/send transport (response-for msg cause)))
-    (t/send transport (response-for msg :status :no-error)))
+  (letfn [(get-ex [] (if storage-key
```

Comment:
There's no need to factor this out as function if it's only used once. I would inline this, and replace the whole `and+not+if` construction below with a `cond`.